src/node_buffer.cc

```diff
@@ -844,6 +844,10 @@ void IndexOfString(const FunctionCallbackInfo<Value>& args) {
 
     if (IsBigEndian()) {
       StringBytes::InlineDecoder decoder;
+      if (!StringBytes::IsValidString(needle, enc)) {
```

Comment:
This is actually a noop here since we already know `enc == UCS2` and `IsValidString` only checks HEX. Can you move the check earlier, above `  if (enc == UCS2)`?

Reply:
Fixed.